docs/DevGuide/TensorExpressions.md

```diff
@@ -0,0 +1,411 @@
+```
+
+## More features {#te_more_features}
+
+### Assigning to a number {#te_assigning_to_a_number}
+You can assign a number (e.g. `double`) to a `Tensor` of any rank to fill all
+components with that value:
+
+\f$L_{ab} = -1\f$
+```
+tnsr::ab<double, 3> L{};
+tenex::evaluate<ti::a, ti::b>(make_not_null(&L), -1.0);
+```
+If the data type of your LHS `Tensor` is a Blaze vector type
+(e.g. `DataVector`), the `Tensor` must be sized before calling
+\ref tenex::evaluate "evaluate" because there is no sizing information (from a
+`Tensor` component) in the RHS expression:
+```
+// construct LHS tensor with size 5 DataVector
+tnsr::ab<DataVector, 3> L{DataVector(0.0, 5)};
+tenex::evaluate<ti::a, ti::b>(make_not_null(&L), -1.0);
+```
+
+See [supported number types](#te_data_type_support).
+
+### Using spatial and time indices on spacetime indices {#te_spatial_time_index}
+If a `Tensor` has spacetime indices, you can use generic spatial indices and
+concrete time indices to refer to a subset of the components, as we see in
+literature.
+
+Lapse \f$\alpha\f$ computed from the spacetime metric \f$g_{ab}\f$ and spatial
+metric \f$\gamma_{ij}\f$:
+
+\f$\alpha = \sqrt{\gamma^{ij} g_{jt} g_{it} - g_{tt}}\f$
+```
+// spatial_metric is type tnsr::ii<DataVector, 3> and spacetime_metric is type
+// tnsr::aa<DataVector, 3>
+auto lapse = tenex::evaluate(
+      sqrt(spatial_metric(ti::I, ti::J) * spacetime_metric(ti::j, ti::t) *
+               spacetime_metric(ti::i, ti::t) -
+           spacetime_metric(ti::t, ti::t)));
+```
+
+### Assigning subsets of tensor components {#te_assigning_subsets_of_components}
+Related to the previous example, you can also use generic spatial indices and
+concrete time indices for the spacetime indices of the LHS `Tensor` to assign
+subsets of the LHS `Tensor`'s components.
+
+Spacetime metric \f$g_{ab}\f$ computed from the lapse \f$\alpha\f$, shift
+\f$\beta^I\f$, and spatial metric \f$\gamma_{ij}\f$:
+
+\f{align}{
+  g_{tt} &= - \alpha^2 + \beta^m \beta^n \gamma_{mn} \\
+  g_{ti} &= \gamma_{mi} \beta^m  \\
+  g_{ij} &= \gamma_{ij}
+\f}
+```
+// spatial_metric is type tnsr::ii<DataVector, 3>, shift is type
+// tnsr::I<DataVector, 3>, and lapse is type Scalar<DataVector>
+
+tnsr::aa<DataVector, 3> spacetime_metric{};
+tenex::evaluate<ti::t, ti::t>(
+    make_not_null(&spacetime_metric),
+    -lapse() * lapse() + shift(ti::M) * shift(ti::N) *
+                spatial_metric(ti::m, ti::n));
+tenex::evaluate<ti::t, ti::i>(
+    make_not_null(&spacetime_metric),
+    spatial_metric(ti::m, ti::i) * shift(ti::M));
+tenex::evaluate<ti::i, ti::j>(
+    make_not_null(&spacetime_metric), spatial_metric(ti::i, ti::j));
+```
+
+### Using the LHS Tensor in the RHS expression {#te_using_lhs_tensor_in_rhs}
+You can use the LHS `Tensor` in the RHS expression to emulate operations like
+`+=`, `*=`, etc. For example, say you would like to emulate the following:
+```
+// pseudocode
+L_ab = R_ab
+L_ab += + 2.0 * S_ba
```

Comment:
I think this example is a little confusing because of the swapped indices on $S_{ba}$.  Below you say that the LHS tensor must have the same index ordering as the RHS. In the "good" example you have `L(ti::a, ti::b) + 2.0 * S(ti::b, ti::a)` which have opposite ordering and in the "bad" example you have `L(ti::b, ti::a) + 2.0 * S(ti::b, ti::a)` which have the same ordering.

I think making S be $S_{ab}$ would be clearer.

Reply:
This was a very useful comment to me because the constraint here is only that `L` must have the same ordering on both sides - other tensors' indices can be doing whatever they want, and I think that's precisely why I left `S` with the opposite order, to demonstrate that `L` is the only one that needs to match the LHS. I see now how the sentence makes it seem like the whole RHS needs to have some ordering, so I'll fix the wording to make it more clear that it's only for `L` and leave the ordering of `S`, if that works for you

Reply:
As long as it's clear, that's good with me